### PR TITLE
Standalone HF Sniff (and ULC/NTAG/ULEV1 pwd storing)

### PR DESCRIPTION
**Pull request by bogiton**

### CODE REVIEW

File: armsrc/Standalone/hf_bog.c [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+	
+	SpinDelay(500);
+
+	/* This is actually copied from SniffIso14443a */
+	
+	iso14443a_setup(FPGA_HF_ISO14443A_SNIFFER);
+	
+	// Allocate memory from BigBuf for some buffers
+	// free all previous allocations first
+	BigBuf_free(); BigBuf_Clear_ext(false);
```
Comment (RfidResearchGroup) [inline armsrc/Standalone/hf_bog.c]: will this free not cancel your previous malloc?

File: armsrc/Standalone/hf_bog.c [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+// main code for standalone HF Sniff (and ULC/NTAG/ULEV1 pwd storing)
+//-----------------------------------------------------------------------------
+#include "hf_bog.h"
+
+#define DELAY_READER_AIR2ARM_AS_SNIFFER (2 + 3 + 8) 
+#define DELAY_TAG_AIR2ARM_AS_SNIFFER (3 + 14 + 8)
+
+// Maximum number of auth attempts per standalone session
+#define MAX_PWDS_PER_SESSION 20
+
```
Comment (RfidResearchGroup) [inline armsrc/Standalone/hf_bog.c]: only 20?   20*4 == 80 ..  Try 256/4= 64?

File: armsrc/Standalone/hf_bog.c [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+	Dbprintf("Stopped sniffing");
+	
+	SpinDelay(200);
+	
+	// Write stuff to flash
+	if (auth_attempts > 0) {
+		if (MF_DBGLEVEL > 1) Dbprintf("[!] Authentication attempts = %u", auth_attempts);
+		
+		// Setting the SPI Baudrate to 48MHz to avoid the bit-flip issue (https://github.com/RfidResearchGroup/proxmark3/issues/34)
+		FlashmemSetSpiBaudrate(48000000);
```
Comment (RfidResearchGroup) [inline armsrc/Standalone/hf_bog.c]: Don't you need to set this only once?

Comment (RfidResearchGroup) [reply armsrc/Standalone/hf_bog.c]: ok,  saw the other call.

**Comment by RfidResearchGroup** [comment]
I guess I need to figure out this reviews stuff...

File: armsrc/Standalone/hf_bog.c (hunk 2 above)
Comment (bogiton) [reply armsrc/Standalone/hf_bog.c]: Yep, I increased it.

File: armsrc/Standalone/hf_bog.c (hunk 1 above)
Comment (bogiton) [reply armsrc/Standalone/hf_bog.c]: Actually I think that this just zeroes out the Bigbuf, it doesn't leave dangling pointers, but I moved it after the free call just to be safe.

**Comment by bogiton** [comment]
Line in changelog and description added.

File: armsrc/Standalone/hf_bog.c (hunk 1 above)
Comment (RfidResearchGroup) [reply armsrc/Standalone/hf_bog.c]: also resets the pointer,  meaning next call to malloc will point to same area...

**Comment by RfidResearchGroup** [comment]
Awesome!

**Pull request merged**